octodns/provider/route53.py

```diff
@@ -465,6 +750,75 @@ def _load_records(self, zone_id):
 
         return self._r53_rrsets[zone_id]
 
+    def _data_for_dynamic(self, name, _type, rrsets):
+        pools = defaultdict(lambda: {'values': []})
+        # Data to build our rules will be collected here and "converted" into
+        # their final form below
+        rules = defaultdict(lambda: {'pool': None, 'geos': []})
+        # Base/empty data
+        data = {
+            'dynamic': {
+                'pools': pools,
+                'rules': [],
+            }
+        }
+
+        # For all the rrsets that comprise this dynamic record
+        for rrset in rrsets:
+            name = rrset['Name']
+            if '-pool.' in name:
+                # This is a pool rrset
+                pool_name = name.split('.', 1)[0][9:-5]
```

Comment:
is `[9:-5]` stripping `_octodns-...-pool`? Might be good to have a comment to note this, took me a sec :smile:

Reply:
Pulled this duplicated parsing bit out into a helper function with the doc and doc'd the singular case where it's parsing SetIdentifier below.